flake.nix

```diff
@@ -4,59 +4,49 @@
   inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.05";
   inputs.flake-utils.url = "github:numtide/flake-utils";
 
-  outputs = { self, nixpkgs, flake-utils }: (flake-utils.lib.eachDefaultSystem (system: let
-    pkgs = import nixpkgs { inherit system; overlays = [ self.overlay ]; };
-  in {
-    packages = {
-      inherit (pkgs.haskellPackages) Agda;
-
-      # TODO agda2-mode
-    };
-
-    packages.default = self.packages.${system}.Agda;
-
-    devShells.default = pkgs.haskellPackages.shellFor {
-      packages = ps: with ps; [ Agda ];
-      nativeBuildInputs = with pkgs; [
-        cabal-install
-        haskell-language-server
-        haskellPackages.fix-whitespace
-
-        # documentation
-        (python3.withPackages (ps: with ps; [
-          sphinx
-          sphinx_rtd_theme
-        ]))
-      ];
-    };
-  })) // {
-    overlay = final: prev: {
-      haskellPackages = prev.haskellPackages.override {
-        overrides = self.haskellOverlay;
+  outputs = { self, nixpkgs, flake-utils }:
+    # Support all the OSes
+    (flake-utils.lib.eachDefaultSystem (system: let
+
+    pkgs = nixpkgs.legacyPackages.${system};
+    hlib = pkgs.haskell.lib.compose;
+    hpkgs = pkgs.haskellPackages;
+
+    # The `agda` and `agda-mode` programs, built with `cabal build`
+    # (and GHC & Haskell libraries from the nixpkgs snapshot)
+    agda-pkg = hpkgs.developPackage {
+        root = ./.;
+        modifier = hlib.dontCheck;
+        # TODO Make check phase work
+        # At least requires:
+        #   Setting AGDA_BIN (or using the Makefile, which at least requires cabal-install)
+        #   Making agda-stdlib available (or disabling the relevant tests somehow)
       };
-    };
-
-    haskellOverlay = final: prev: let
```

Comment:
Why are we removing the overlay? This is needed to properly use a HEAD version of agda in one's own config.

Reply:
Arguably the proper solution here is to allow `packages` to be parameterized by a `pkgs` set or similar. But without that, overlays are the mechanism we have to compose rather than inherit!

Reply:
Thanks for the explanation!
c0dc559 adds an overlay again, do you have an overlay-consuming project we can test it on?

Reply:
I don't have one currently, I only use this in my own config when I need something from Agda HEAD. But if we're just using the easyOverlay stuff, that's probably taken care of by their upstream testing right?

Reply:
> But if we're just using the easyOverlay stuff, that's probably taken care of by their upstream testing right?

Provided that I set up the `easyOverlay` correctly, which I am not sure about.

Reply:
ff5a7bb marks the overlay as EXPERIMENTAL.
I propose we merge this without waiting for overlay tests, and fix overlay bugs if/when they are reported.